src/modules/base.ts

```diff
@@ -1,119 +1,137 @@
 /**
  * Base module for all API modules
  *
- * @module modules/base
- *
  * @since 1.0.0
  */
 import commsUtils from '../utils/comms';
-import {name as PROTOCOL_AGENT_NAME, version as PROTOCOL_AGENT_VERSION} from '../../package.json';
 
 const PROTOCOL_NAME = 'purecloud-client-apps';
 
-const ACTION_NAME_KEY = 'action';
-const PROTOCOL_NAME_KEY = 'protocol';
-const PROTOCOL_AGENT_NAME_KEY = 'protocolAgentName';
-const PROTOCOL_AGENT_VERSION_KEY = 'protocolAgentVersion';
+interface ProtocolDetails {
+    name: string;
+    agentName: string;
+    agentVersion: string;
+}
+
+export interface SDKMessagePayload {
+    action: string;
+    hook?: string;
```

Comment:
If we defined a base interface of InboundSDKMessagePayload here, could we extend it in the classes using listeners so they could add their custom payload details and the base wouldn't need to be aware of them?

Or, would that require making the base class a generic?  If so, we can certainly defer that work as this is the only property/usage as of now and YAGNI.  If we defer, might then make sense to split listeners out as a separate generic class and let downstream classes just compose in the message behavior.

Reply:
Yeah, this is good to consider. It shouldn't require us to make the base generic, so long as the base doesn't need to access these additional properties. I can certainly clean this up a bit now that I understand the structure better.

Reply:
Going to correct myself here. It seems like we actually probably do want to go the route of making the base class generic as that will allow us to assert that certain properties are actually on the payload object. My initial thought was that since `hook` was an optional property that it wouldn't require us to make the base class generic. However, after a second go-around in my head, we more than likely will want to assert that it is a required property, and to get TS to be happy with that we will need to make the base generic. So maybe we should punt on this for now. Either way, good to consider.